src/verify.rs

```diff
@@ -462,7 +463,7 @@ pub fn verify(result: &mut VerifyResult,
             let sref = segments2.segment(id);
             if let Some(old_res) = old_res_o {
                 if old_res.scope_usage.valid(&nset, &scope) &&
-                   ptr_eq::<Segment>(&old_res.source, sref.segment) {
+                   ptr_eq(&old_res.source, sref.segment) {
```

Comment:
I'm always using a type with `ptr_eq` because, with `Deref` instances in play, it's not always easy to tell _what_ is being compared.  It'd be very bad if that were resolved as `ptr_eq::<Arc<Segment>>`, as it would completely defeat the recalculation check.